virttest/virsh.py

```diff
@@ -1634,12 +1645,19 @@ def detach_device(domainarg=None, filearg=None,
     :param domain_opt: Option to --domain parameter
     :param file_opt: Option to --file parameter
     :param flagstr: string of "--force, --persistent, etc."
+    :param wait_remove_event: wait until device_remove event comes
+    :param event_timeout: timeout for virsh event command
     :param dargs: standardized virsh function API keywords
     :return: CmdResult instance
     """
-    return _adu_device("detach-device", domainarg=domainarg, filearg=filearg,
-                       domain_opt=domain_opt, file_opt=file_opt,
-                       flagstr=flagstr, **dargs)
+    detach_cmd_rv = _adu_device("detach-device", domainarg=domainarg, filearg=filearg,
+                                domain_opt=domain_opt, file_opt=file_opt,
+                                flagstr=flagstr, **dargs)
+    if wait_remove_event:
+        event_cmd = "event --domain %s --event device-removed --timeout %d" % (domainarg, event_timeout)
+        command(event_cmd, **dargs)
```

Comment:
these 2 lines are repeated many times in this patch, it's better to call event function I think

Reply:
yeah. That's tidier. I'll update it.